lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2440,8 +2439,8 @@ instance LiftHandler ErrSignPayment where
     handler = \case
         ErrSignPaymentMkTx e -> handler e
         ErrSignPaymentNoSuchWallet e -> (handler e)
-            { errHTTPCode = 410
```

Comment:
Ok, so `Gone` would be a viable option if we know the wallet existed beforehand. But I guess we don't?

Reply:
So, all uses of this type are as follows

```hs
    withRootKey @_ @s ctx wid pwd ErrSignPaymentWithRootKey $ \xprv scheme -> do
        let pwdP = preparePassphrase scheme pwd
        mapExceptT atomically $ do
            cp <- withExceptT ErrSignPaymentNoSuchWallet $ withNoSuchWallet wid $
                readCheckpoint (PrimaryKey wid)
            (cs', s') <- assignChangeAddressesForSelection
                argGenChange cs (getState cp)
            withExceptT ErrSignPaymentNoSuchWallet $
                putCheckpoint (PrimaryKey wid) (updateState s' cp)

            let keyFrom = isOwned (getState cp) (xprv, pwdP)
            let rewardAcnt = mkRewardAccount (xprv, pwdP)

            (tx, sealedTx) <- withExceptT ErrSignPaymentMkTx $ ExceptT $
                pure $ mkStdTx tl rewardAcnt keyFrom txExp md cs'

            (time, meta) <- liftIO $ mkTxMeta ti (currentTip cp) s' tx cs' txExp
            return (tx, meta, time, sealedTx)
```

That means we can only hit this code-path if we found a root key for the given wallet id. Does that imply the wallet existed?

Reply:
Gone _made sense_ in the context of concurrent request. When one creates a transaction, there are multiple back-and-forth with the database in separate db transactions. In principle, the wallet could disappear between any of these db transaction might there be a concurrent request deleting the wallet at the same time.

Arguably, this is veeeeeeeery edgy, and from a user perspective doesn't matter much. So changing to 404 sounds like a good move.